tools/astarte_e2e/lib/astarte_e2e/service_notifier.ex

```diff
@@ -63,6 +65,28 @@ defmodule AstarteE2E.ServiceNotifier do
     end
   end
 
+  defp update_failures_before_alert(data) do
+    %{data | failures_before_alert: data.failures_before_alert - 1}
+  end
+
+  defp set_failures_before_alert_to(data, num) do
+    %{data | failures_before_alert: num}
+  end
+
+  defp reset_failures_before_alert(data) do
+    %{data | failures_before_alert: Config.failures_before_alert!()}
+  end
+
+  defp update_failure_id(data, event_id) do
+    %{data | failure_id: event_id}
+  end
+
+  defp get_failure_id(%{failure_id: failure_id} = _data), do: failure_id
+
+  defp reset_failure_id(data) do
+    %{data | failure_id: @default_failure_id}
```

Comment:
I don't see why you have to implement these instead of just using functions from the `Map` module